### PR TITLE
Update Edge and IE typed arrays compat data

### PR DESCRIPTION
**Pull request by Elchi3**
This is a test PR to measure how long it takes to update a chunk of Edge JS compat data. It updates 45 data points and I've spend 1h on it. Please let me know how long you spend reviewing.

Most of these things were figured out by testing in Saucelabs or by making sense of the data that was already provided. If have used several commits to record my thought process.

### CODE REVIEW

**Comment by vinyldarkscratch** [comment]
Haven't fully started reviewing this just yet, but I noticed some parts were marked as Edge 14, when IE is `null`.  I'm thinking, perhaps we should assume that IE is `false` in those scenarios, and update the data accordingly?  (We could make it a separate PR of course!)

**Comment by Elchi3** [comment]
Thanks for starting the review! IE is probably false then, yes, but I think I would do IE separately. 
It's hard to draw the line sometimes. I thought this PR was large enough already.

**Comment by vinyldarkscratch** [comment]
Ah, that makes sense!  I'll write up a follow-up PR to help reflect that information.  :wink:

I'm also assuming that when Edge is set to 12 based upon SauceLabs tests, but IE is `null`, you've found that IE had support in some version.  Am I correct in this assumption?

**Comment by Elchi3** [comment]
I think so, yes. I just wanted to make this PR just about Edge but I'm not actually sure if that is helpful or not. If it means we're spending more time, we should update Edge and IE together? I'm looking into how we could spend less time on getting real values. What do you think?

**Comment by vinyldarkscratch** [comment]
During my work on the CSS real values, I often tackled both IE and Edge together.  Personally, I feel that we should tackle JavaScript with that same idea.  I think that it will help reduce the amount of time obtaining real values, and overall a little more efficient, as we'd be spending a little more effort on each individual feature, and less effort doing another pass over everything.

**Comment by Elchi3** [comment]
I've added IE updates now.
I wonder if https://github.com/mdn/browser-compat-data/issues/4772 is organized well enough for us then, or if we should chunk (and estimate) the work differently.

File: javascript/builtins/TypedArray.json [hunk 1]
```diff
@@ -964,7 +965,7 @@
                 "version_added": null
               },
               "edge": {
-                "version_added": null
+                "version_added": "14"
```
Comment (vinyldarkscratch) [inline javascript/builtins/TypedArray.json]: IE was set to "10" in #2331, however we both know this is not possible as the feature was added in Edge 14 (I've double-checked this just to be on the safe side).  Perhaps we should update that value here?

File: javascript/builtins/TypedArray.json [hunk 2]
```diff
@@ -2059,7 +2060,7 @@
                 "version_added": true
               },
               "edge": {
-                "version_added": null
+                "version_added": "12"
```
Comment (vinyldarkscratch) [inline javascript/builtins/TypedArray.json]: Looks like this was also implemented in IE 10.

**Comment by Elchi3** [comment]
Great catches! Updated the two values.
